### ISSUE TITLE
Native module cannot be null

### ISSUE THREAD

**Issue by iamclaytonray** [expo/expo#593]
So I boilerplated a quick little app to throw on here for you guys. I've been trying to figure this out all day. Also, I am not 100% sure this is related to Expo or another library, but it seems to be the case when I import anything from Expo. I keep getting an error saying: "Native module cannot be null" as seen in the screenshot below. I've pretty much copied/pasted code from creating a new Expo project into a `react-native init`'ed app. I've followed all of the guidelines/steps and have had no luck at all. <a href="https://github.com/iamclaytonray/AwesomeApp">Here's my repo with my current code.</a>

![simulator screen shot sep 4 2017 6 10 21 pm](https://user-images.githubusercontent.com/16495816/30039242-57f91e6a-919c-11e7-91f7-b84d3f4127be.png)

**Comment by iamclaytonray** [comment]
If this is unrelated to Expo, can you point me to where I should be going? It could have something to do with react-native-vector-icons but I am really just not sure.

**Comment by dikaiosune** [comment]
Hi! Do you think you could provide a minimal repro using [Snack](https://snack.expo.io)? The repository you linked has a bunch of different things in it, would be much easier to troubleshoot with less code.

**Comment by iamclaytonray** [comment]
I'll throw it up there in a couple of hours, sure. Just as a side note, you could pretty much ignore everything besides `/src` and `index.ios.js` on that repo. All of the other stuff is what you get when you run `react-native init SomeProjectName`

**Comment by iamclaytonray** [comment]
* Which has not been modified at all

**Comment by fatwhale** [comment]
I came into this problem as well today, and couldn't find a way to solve. Did you solved this ? @iamclaytonray / Or is there any findings @dikaiosune

**Comment by iamclaytonray** [comment]
I have not found a solution.

**Comment by iamclaytonray** [comment]
Sorry, @dikaiosune, I got super busy. I'll try to do it today. But, if it is a native module error, would it not be better to debug with the Objective-C code? `ios/` and `android/` has not been modified at all, so you wouldn't have to look at any of the code there.

Ironically, I actually deleted the repo. I'll have to create a new one.

**Comment by iamclaytonray** [comment]
@fatwhale - If you find a solution before either of us do, can you drop a comment with your fix?

**Comment by ide** [comment]
Does any of your JS require “expo”? It looks like the error is failing while trying to set up Expo’s Location API. If so you’ll need Expo’s native code. Using a bare React Native project won’t work.

**Comment by iamclaytonray** [comment]
No, I wasn't using anything with Maps/Location. I did add Expo as a dependency. Do I have to use the Expo IDE to actually use Expo? That wouldn't make sense to me but that's kind of what it sounds like. Thanks!

**Comment by iamclaytonray** [comment]
I can't quite remember what I was using. I think it was Font/Icon. Shouldn't have deleted the repo 🙄

**Comment by ide** [comment]
Expo isn’t an IDE, it’s a development platform. The Expo SDK you get from npm is the JS API that lets you use features that are built into Expo.

**Comment by iamclaytonray** [comment]
I guess that pretty much answers my question then. I'll change my method for dealing with Expo. Thanks!

**Comment by iamclaytonray** [comment]
And I didn't mean that the IDE & SDK were bound together. I just know that the IDE generates (or the CLI) a RN app very different from `react-native init AwesomeProject`. That's what I was talking about

**Comment by fatwhale** [comment]
Agree, this problem only popped out after I ejected my project, and reinstalled expo in the ejected directory, as by default the expo module will be removed when ejection.

After ejection, initially, iIgot the error complaining about the line **"linkingUrl: ExponentConstants.linkingUri,"** in the **Constans.js**, so I commented it out. Then the problem above popped out.

As for the line **"linkingUrl: ExponentConstants.linkingUri,"**, if you use expo in 'create-react-native-app', it won't be there though. But when you use react-native init/eject. you will find this line.

As for the problem above, so far I still couldn't solve it but I find that, module 'expo' and '@expo/vector-icons' got me into the above situation, whereas, using @expo/react-native-action-sheet is fine.
